### PR TITLE
(CDAP-12466) Added a simple fencing mechanism to TMS

### PR DESCRIPTION
**Pull request by chtyim**
- Just delaying making the service becomes available

### CODE REVIEW

**Comment by chtyim** [comment]
Build https://builds.cask.co/browse/CDAP-RUT1296

File: cdap-tms/src/main/java/co/cask/cdap/messaging/distributed/LeaderElectionMessagingService.java [hunk 1]
```diff
@@ -195,16 +205,46 @@ private HBaseTableFactory getHBaseTableFactory(TableFactory tableFactory) {
...
   }
 
+  private void fencingStart(final DelegateService service) {
+    Runnable runnable = new Runnable() {
+      @Override
+      public void run() {
+        service.startAndWait();
+        // If failed to mark the service to become available, this means the follower() call happened before this,
+        // so just go ahead and shutdown the service.
+        if (!delegate.attemptMark(service, true)) {
```
Comment (gokulavasan) [inline cdap-tms/src/main/java/co/cask/cdap/messaging/distributed/LeaderElectionMessagingService.java:230]: javadoc seems to indicate that the attemptMark method might fail spuriously. Would that affect the behavior?

**Comment by gokulavasan** [comment]
One question. Rest LGTM 👍

File: cdap-tms/src/main/java/co/cask/cdap/messaging/distributed/LeaderElectionMessagingService.java (hunk 1 above)
Comment (chtyim) [reply cdap-tms/src/main/java/co/cask/cdap/messaging/distributed/LeaderElectionMessagingService.java:230]: fail when there are concurrent modification and it returns `false`, which we handled.

**Comment by chtyim** [comment]
Build passed

**Comment by gokulavasan** [comment]
LGTM 👍

**Pull request merged**